chain.go

```diff
@@ -205,6 +205,41 @@ func (cc *Conn) ListChains() ([]*Chain, error) {
 	return chains, nil
 }
 
+// GetChain gets a chain by name
+func (cc *Conn) GetChain(name string) (*Chain, error) {
+	conn, err := cc.dialNetlink()
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	msg := netlink.Message{
+		Header: netlink.Header{
+			Type:  netlink.HeaderType((unix.NFNL_SUBSYS_NFTABLES << 8) | unix.NFT_MSG_GETCHAIN),
+			Flags: netlink.Request | netlink.Dump,
+		},
+		Data: extraHeader(uint8(unix.AF_UNSPEC), 0),
+	}
+
+	response, err := conn.Execute(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, m := range response {
+		c, err := chainFromMsg(m)
+		if err != nil {
+			return nil, err
+		}
+		if c.Name == name {
```

Comment:
Here and for GetTable(), you’re still requesting all chains/tables from the kernel, and then filter after receiving them.

As discussed previously, it seems better to me to only request the chains/tables you need.

You’re already not setting NLM_F_ECHO, so perhaps all that’s needed is to specify the name/family attributes when sending the request to the kernel? Have you tried that?

Reply:
ok, i will have a try, nice suggestions